stages/org.osbuild.gcp.guest-agent.conf

```diff
@@ -0,0 +1,269 @@
+    "minProperties": 1,
+    "properties": {
+      "setup": {
+        "type": "boolean",
+        "description": "Skips network interface setup."
+      },
+      "ip_forwarding": {
+        "type": "boolean",
+        "description": "Skips IP forwarding."
+      },
+      "dhcp_command": {
+        "type": "string",
+        "description": "Path for alternate dhcp executable used to enable network interfaces."
+      }
+    }
+  }
+},
+"additionalProperties": false,
+"required": ["config"],
+"description": "Configure GCP guest-agent.",
+"properties": {
+  "config_scope": {
+    "type": "string",
+    "description": "Create distro-wide or instance-specific configuration.",
+    "enum": ["distro", "instance"],
+    "default": "distro"
+  },
+  "config": {
+    "type": "object",
+    "additionalProperties": false,
+    "description": "GCP guest-agent configuration.",
+    "minProperties": 1,
+    "properties": {
+      "Accounts": {
+        "$ref": "#/definitions/Accounts"
+      },
+      "Daemons": {
+        "$ref": "#/definitions/Daemons"
+      },
+      "InstanceSetup": {
+        "$ref": "#/definitions/InstanceSetup"
+      },
+      "IpForwarding": {
+        "$ref": "#/definitions/IpForwarding"
+      },
+      "MetadataScripts": {
+        "$ref": "#/definitions/MetadataScripts"
+      },
+      "NetworkInterfaces": {
+        "$ref": "#/definitions/NetworkInterfaces"
+      }
+    }
+  }
+}
+"""
+
+def option_value_to_str(value):
+    """
+    Convert allowed types of option values to string.
+
+    The instance_configs.cfg allows comma seperated string lists as an option value.
+    """
+    if isinstance(value, list):
+        value = ", ".join(value)
+    elif isinstance(value, bool):
+        value = "true" if value else "false"
+    elif not isinstance(value, str):
+        value = str(value)
+    return value
+
+def make_section(cfg, name, settings):
+    if not cfg.has_section(name):
+        cfg.add_section(name)
+
+    for key, value in settings.items():
+        val = option_value_to_str(value)
+        cfg.set(name, key, val)
+
+
+def main(tree, options):
+    scope = options.get("config_scope")
+    config = options.get("config")
+
+    instance_config_dir = f"{tree}/etc/default"
+    os.makedirs(instance_config_dir, exist_ok=True)
+
+    if scope == "distro":
+        filepath = instance_config_dir + "/instance_configs.cfg.distro"
+    else:
+        filepath = instance_config_dir + "/instance_configs.cfg"
+
+    parser = iniparse.SafeConfigParser()
+
+    try:
+        with open(filepath, "r", encoding="utf-8") as f:
+            parser.readfp(f)
+    except FileNotFoundError:
+        print(f"Warning: guest-agent configuration file '{filepath}' does not exist, will create it.")
+        os.makedirs(instance_config_dir, exist_ok=True)
+
+    for section_id, section_content in config.items():
+        make_section(parser, section_id, section_content)
+
+    with open(filepath, "w") as f:
```

Comment:
This is going to trigger a warning in the future about the missing `encoding="utf-8"`